src/forms/form.rs

```diff
@@ -0,0 +1,355 @@
+    }
+
+    fn get_height(&self) -> u32 {
+        self.height
+    }
+    fn get_min_height(&self) -> u32 {
+        3
+    }
+
+    fn resize(&mut self, w: u32, h: u32) {
+        self.height = h;
+        self.screen.resize(w, self.screen.get_height());
+        for (_, field) in self.fields.iter_mut() {
+            field.resize(w, field.get_height());
+        }
+    }
+
+    fn handle_event(&mut self, event: &Event) {
+        for (_, field) in self.fields.iter_mut() {
+            field.handle_event(event);
+        }
+        if let Event::Key(KeyEvent { code, modifiers: _ }) = event {
+            match code {
+                KeyCode::Enter => {
+                    self.index = (self.index + 1).clamp(0, self.fields.len());
+                    self.update_active_field();
+                }
+                KeyCode::Tab => {
+                    self.index = (self.index + 1).clamp(0, self.fields.len() - 1);
+                    self.update_active_field();
+                }
+                KeyCode::BackTab => {
+                    self.index =
+                        (self.index as i64 - 1).clamp(0, self.fields.len() as i64) as usize;
+                    self.update_active_field();
+                }
+                KeyCode::PageDown => {
+                    self.scroll(1);
+                }
+                KeyCode::PageUp => {
+                    self.scroll(-1);
+                }
+                _ => {}
+            }
+        }
+    }
+
+    fn set_active(&mut self, active: bool) {
+        self.active = active;
+        self.update_active_field();
+    }
+
+    fn is_active(&self) -> bool {
+        self.active
+    }
+
+    fn validate(&self, validation_result: &mut FormValidationResult) {
+        for (_, field) in self.fields.iter() {
+            field.validate(validation_result);
+        }
+        self.self_validate(validation_result);
+    }
+
+    fn get_output(&self) -> FormValue {
+        let mut output: HashMap<String, FormValue> = HashMap::new();
+        for (name, field) in self.fields.iter() {
+            output.insert(name.to_string(), field.get_output());
+        }
+        FormValue::Map(output)
+    }
+
+    fn set_options(&mut self, options: FormOptions) {
+        self.dirty = true;
+        self.options = options;
+    }
+
+    fn get_options(&self) -> &FormOptions {
+        &self.options
+    }
+
+    fn draw(&mut self, tick: usize) -> &Screen {
+        let mut total_height = 1;
+        for (_, field) in self.fields.iter_mut() {
+            total_height += field.get_height();
+            if field.display_label() {
+                total_height += 1;
+            }
+        }
+        if self.screen.get_height() != total_height {
+            self.screen.resize(self.screen.get_width(), total_height);
+        }
+        if self.dirty {
+            let padding = if self.options.style.border.is_some() {
+                1
+            } else {
+                0
+            };
```

Comment:
In my opinion `let padding = self.options.style.border.is_some() as i32;` would look cleaner.